File: clojure/bob/bob.clj
(ns bob)

(defn re-test [pattern phrase]
  (let [matches (re-matches pattern phrase)
        match (if (vector? matches)
                (first matches)
                matches)]
    (if (= match phrase)
      true
      false)))

(defn response-for [phrase]
  (let [question? #(re-test #"[A-Z0-9](?:[a-z\s]*)\?$" %)
        yelling?  #(re-test #"([\d\W]*?)(?:[A-Z\s]{2,})[^a-z]*" %)
        blank?    #(re-test #"^\s*$" %)]
    (cond (blank? phrase)
          "Fine. Be that way!"
          (yelling? phrase)
          "Whoa, chill out!"
          (question? phrase)
          "Sure."


          :else
          "Whatever.")))

File: objective-c/bob/Bob.m
//
//  Bob.m
//  
//
//  Created by Jason Lewis on 1/8/15.
//
//

#import <Foundation/Foundation.h>
#import "Bob.h"

@implementation Bob

NSString *shoutResponse = @"Whoa, chill out!";
NSString *questionResponse = @"Sure.";
NSString *blankResponse = @"Fine. Be that way!";
NSString *defaultResponse = @"Whatever.";

- (BOOL)isShout:(NSString *)phrase {
    if ([[phrase uppercaseString] isEqualToString:phrase ]) {
        return YES;
    } else {
        return NO;
    }
}


- (NSString*) hey: (NSString*) phrase {
    
}

@end
